packages/csharp/src/components/Class.tsx

```diff
@@ -0,0 +1,110 @@
+import * as core from "@alloy-js/core";
+import * as base from "./index.js";
+import * as csharp from "../index.js";
+import * as symbols from "../symbols/index.js";
+
+// properties for creating a class
+export interface ClassProps extends Omit<base.DeclarationProps, "nameKind"> {
+  accessModifier?: csharp.AccessModifier;
+}
+
+// a C# class declaration
+export function Class(props: ClassProps) {
+  const name = csharp.useCSharpNamePolicy().getName(props.name, "class");
+  const scope = symbols.useCSharpScope();
+
+  const thisClassSymbol = scope.binder.createSymbol<symbols.CSharpOutputSymbol>({
+    name: name,
+    scope,
+    refkey: props.refkey ?? core.refkey(props.name),
+  });
+
+  // this creates a new scope for the class definition.
+  // members will automatically "inherit" this scope so
+  // that refkeys to them will produce the fully-qualified
+  // name e.g. Foo.Bar.
+  const thisClassScope = symbols.createCSharpMemberScope(scope.binder, scope, thisClassSymbol, "class");
+
+  return <core.Declaration symbol={thisClassSymbol}>
+      {csharp.getAccessModifier(props.accessModifier)}class <base.Name />{props.children ? (
```

Comment:
I might add `{!props.children && ";"}` at the start to avoid the massive ternary.